lib/Api.php

```diff
@@ -40,25 +40,17 @@ private function __changestatus()
         $table   = rex_post('table', 'string');
         $sql     = rex_sql::factory();
 
-        $sql->setTable($table)->setValue('status', $status)->setWhere(['id' => $data_id]);
+        $dataset = \rex_yform_manager_dataset::get($data_id, $table);
+        $dataset->setValue('status', $status);
+
         try {
-            $sql->update();
+            $dataset->save();
```

Comment:
@tbaddade habe ich das falsch im Blick oder müsste ein gespeichertes dataset nicht automatisch die URL des Datensatzes oder der Datensätze in URL2 aktualisieren? 

Im Moment bleibt die URL, obwohl ich den Datensatz offline stelle und das im Profil richtig eingestellt ist.

Reply:
> oder müsste ein gespeichertes dataset nicht automatisch die URL des Datensatzes oder der Datensätze in URL2 aktualisieren?

Das sollte es. Aber anscheinend wird der EP `YFORM_DATA_UPDATED` nicht ausgelöst?

Reply:
der EP wird abseits dieses Skriptes ordnungsgemäß aufgerufen. Ich habe mal geschaut, ob es mit load:late in der package.yml oder der Abhängigkeit von URL gelöst würde - auch nicht. Ich bin etwas ratlos, warum das hier in diesem Kontext nicht funktioniert.

Reply:
![image](https://user-images.githubusercontent.com/3855487/157844072-6f86fadf-3309-4d3b-a890-578f7154eb68.png)

In Clockwork sehe ich, dass er getriggert wird, aber quasi wirkungslos bleibt. Als würde darin gar nichts passieren.

Reply:
Kann es sein, dass der EP zu früh getriggert wird?

Reply:
Was heißt zu früh? Woran kann man das erkennen?

Ich habe gesehen, dass das URL-Addon sich nur im Backend einhängt, der rex_api-Aufruf geht jedoch über das Frontend. Die Backend-Bedingung im URL-Addon rauszunehmen hat das Problem jedoch auch nicht gelöst. Es scheint ein komplexeres Problem mit verschiedenen Einflussfaktoren sein.

Jetzt verliere ich den Überblick, ob es ein URL (Ladereihenfolge?), YForm (zu frühes triggern?!), YForm Usability (konzeptionell) oder ein Core-Problem (rex_api-Kontext) ist.

Workaround: Vormals in yform_usability 1.3 wurde auf eine Backend-Page weitergeleitet. Das könnte ich wieder so einführen. Sinnvoller erscheint es mir jedoch, wenn der EP das auslöst, was er soll.

Reply:
Vorab: Ich kenne das Addon nicht und weiß somit nicht was genau wie wann passiert.

> Ich habe gesehen, dass das URL-Addon sich nur im Backend einhängt, der rex_api-Aufruf geht jedoch über das Frontend. 

Verstehe ich das richtig, du möchtest den Status eine Datensatzes im Backend beeinflussen? Weshalb geht der rex_api Aufruf dann über das Frontend? 

> Die Backend-Bedingung im URL-Addon rauszunehmen hat das Problem jedoch auch nicht gelöst. Es scheint ein komplexeres Problem mit verschiedenen Einflussfaktoren sein.

Wann und wo wird der rex_api Aufruf ausgelöst? Wo passiert das hier im Code?

Reply:
Meins ist es auch nicht, aber ich habe den Aufbau zumindest inzwischen halbwegs im Blick. Ich mach dir einen Screencast, anders ist es nicht erklärbar, glaube ich.

Reply:
> Verstehe ich das richtig, du möchtest den Status eine Datensatzes im Backend beeinflussen? Weshalb geht der rex_api Aufruf dann über das Frontend?

Das war wohl in einer früheren Version von yform_usability anders, das müsste man die Person fragen, die's umgebaut hat. Die jetzige Lösung basiert sicherlich auf https://friendsofredaxo.github.io/tricks/howto/ajax_rexapi

> Wann und wo wird der rex_api Aufruf ausgelöst? Wo passiert das hier im Code?

Wenn man in einer YForm-Tabelle ist, gibt's dann einen Status-Wechsel (wie bei structure an Artikeln / Kategorien)

Hier ist der Aufruf

https://github.com/FriendsOfREDAXO/yform_usability/blob/f620f48e48002c1041e3deea49ade7a1f505fff7/assets/script.js#L48-L56

Reply:
Damit drehen wir uns im Kreis, denn das ist ja gerade das, was ich in diesem PR hier auf YOrm umstellen möchte:
* das dataset wird fortan statt über rex_sql durch YOrm `$dataset->save();` gespeichert
* dabei wird der EP `YFORM_DATA_UPDATED` getriggert (siehe Unterhaltung weiter oben)
*  trotzdem klappt das nicht. 

Wie du geschrieben hast:

> Kann es sein, dass der EP zu früh getriggert wird?

Das glaube ich auch, nur weiß ich nicht, wie ich darauf Einfluss nehmen kann.

Reply:
Super, danke euch beiden!

Was der Grund war, es so zu lösen - ich weiß es nicht.

Wenns keine negativen Effekte hat, schmeiß ich das raus und fertig.

Reply:
Das hat leider nicht gereicht. Ich breche hier ab. Es bleibt so, wie es ist, ohne YOrm, mangels Alternative.